### ISSUE TITLE
SC2016 false positive

### ISSUE THREAD

**Issue by maxatome** [koalaman/shellcheck#2319]
#### For bugs
- Rule Id SC2016
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
x=toto
any 'qw('"$x"');$z'
```

#### Here's what shellcheck currently says:

```
In x.sh line 4:
any 'qw('"$x"');$z'
             ^----^ SC2016: Expressions don't expand in single quotes, use double quotes for that.

For more information:
  https://www.shellcheck.net/wiki/SC2016 -- Expressions don't expand in singl...
```

#### Here's what I wanted or expected to see:

nothing as there is no error here.

Interesting to note: replace `any` by `perl` and it works...

**Comment by koalaman** [comment]
I've not seen the `any` command before, but you're right, it's clearly meant to accept literal dollar signs. ShellCheck has a hard coded list of commands including `perl` to try to avoid false positives, but it's hard to keep track of all of them. Please ignore it with a disable directive, and don't hesitate to permanently ignore this warning in your `.shellcheckrc` since it's meant to help newbies discover the difference between single and double quotes, and not to stop you from using single quotes correctly. 

I've updated the wiki to be more explicit about this. Thanks!
